File: 1.0/src/c/cz/mg/vulkan/PFN_vkCreateIndirectCommandsLayoutNVX.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCreateIndirectCommandsLayoutNVX_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pCreateInfo, jlong pAllocator, jlong pIndirectCommandsLayout, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkCreateIndirectCommandsLayoutNVX f = (PFN_vkCreateIndirectCommandsLayoutNVX)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkIndirectCommandsLayoutCreateInfoNVX*)jniLongToPointer(pCreateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkIndirectCommandsLayoutNVX*)jniLongToPointer(pIndirectCommandsLayout))
    );
}


jint Java_cz_mg_vulkan_PFNvkCreateIndirectCommandsLayoutNVX_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pCreateInfo, jlong pAllocator, jlong pIndirectCommandsLayout){
    (void)env;
    (void)jc;
    PFN_vkCreateIndirectCommandsLayoutNVX f = (PFN_vkCreateIndirectCommandsLayoutNVX)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkIndirectCommandsLayoutCreateInfoNVX*)jniLongToPointer(pCreateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkIndirectCommandsLayoutNVX*)jniLongToPointer(pIndirectCommandsLayout))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkCreateIndirectCommandsLayoutNVX.java
package cz.mg.vulkan;

public class PFNvkCreateIndirectCommandsLayoutNVX extends VkFunctionPointer {
    public PFNvkCreateIndirectCommandsLayoutNVX() {
    }

    public PFNvkCreateIndirectCommandsLayoutNVX(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkCreateIndirectCommandsLayoutNVX(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCreateIndirectCommandsLayoutNVX(long value) {
        setValue(value);
    }

    public PFNvkCreateIndirectCommandsLayoutNVX(VkInstance instance) {
        super(instance, new VkString("vkCreateIndirectCommandsLayoutNVX"));
    }

    public void call(VkDevice device, VkIndirectCommandsLayoutCreateInfoNVX pCreateInfo, VkAllocationCallbacks pAllocator, VkIndirectCommandsLayoutNVX pIndirectCommandsLayout, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddress(), pCreateInfo != null ? pCreateInfo.getVkAddress() : VkPointer.NULL, pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL, pIndirectCommandsLayout != null ? pIndirectCommandsLayout.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddress());
    }

    private static native void call(long vkaddress, long device, long pCreateInfo, long pAllocator, long pIndirectCommandsLayout, long rval);
}
